src/unitycatalog/ai/utils.py

```diff
@@ -113,3 +141,148 @@ def validate_full_function_name(function_name: str) -> FullFunctionName:
             f"Invalid function name: {function_name}, expecting format <catalog_name>.<schema_name>.<function_name>."
         )
     return FullFunctionName(catalog_name=splits[0], schema_name=splits[1], function_name=splits[2])
+
+
+def uc_type_json_to_pydantic_type(uc_type_json: Union[str, Dict]) -> Type:
+    """
+    Convert Unity Catalog type json to Pydantic type.
+    """
+    if isinstance(uc_type_json, str):
+        type_name = uc_type_json.upper()
+        if type_name in UC_TYPE_JSON_MAPPING:
+            return Union[UC_TYPE_JSON_MAPPING[type_name]]
+        # the type text contains the precision and scale
+        elif type_name.startswith("DECIMAL"):
+            return Union[decimal.Decimal, float]
+        else:
+            raise TypeError(
+                f"Type {uc_type_json} is not supported. Supported "
+                f"types are: {UC_TYPE_JSON_MAPPING.keys()}"
+            )
+    if isinstance(uc_type_json, dict):
+        tpe = uc_type_json["type"]
+        if tpe == "array":
+            element_type = uc_type_json_to_pydantic_type(uc_type_json["elementType"])
+            if uc_type_json["containsNull"]:
+                element_type = Optional[element_type]
+            return Union[List[element_type], Tuple[element_type]]
```

Comment:
```suggestion
            return Union[List[element_type], Tuple[element_type, ...]]
```

`Tuple[element_type]` means a single-element tuple.